packages/backend-defaults/src/alpha/entrypoints/metrics/metricsServiceFactory.ts

```diff
@@ -14,14 +14,28 @@
  * limitations under the License.
  */
 
-import { createFormField } from '@backstage/plugin-scaffolder-react/alpha';
+import { metricsServiceRef } from '@backstage/backend-plugin-api/alpha';
 import {
-  RepoOwnerPicker as Component,
-  RepoOwnerPickerFieldSchema,
-} from '../../components/fields/RepoOwnerPicker';
+  coreServices,
+  createServiceFactory,
+} from '@backstage/backend-plugin-api';
+import { DefaultMetricsService } from './DefaultMetricsService';
 
-export const RepoOwnerPicker = createFormField({
-  component: Component,
-  name: 'RepoOwnerPicker',
-  schema: RepoOwnerPickerFieldSchema,
+/**
+ * Service factory for collecting plugin-scoped metrics.
+ *
+ * @alpha
+ */
+export const metricsServiceFactory = createServiceFactory({
+  service: metricsServiceRef,
+  deps: {
+    pluginMetadata: coreServices.pluginMetadata,
+  },
+  factory: () => {
+    const namespace = 'default';
```

Comment:
Rather than attempting to make everything configurable in a single PR, I elected to provide an implementation that lets us migrate existing metrics to use the new service without introducing any breaking changes. The follow-up would be to implement the `backstage.{plugin_id}` namespacing will migrating existing metrics to their new counterparts. This would introduce a breaking change so we could decide to include a config flag to disable this during alpha phase.